lib_nbgl/src/nbgl_buttons.c

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file nbgl_buttons.c
+ * Implementation of buttons management in NBGL
+ */
+
+#ifndef HAVE_SE_TOUCH
+/*********************
+ *      INCLUDES
+ *********************/
+#include <string.h>
+#include "nbgl_debug.h"
+#include "nbgl_buttons.h"
+#include "nbgl_lns.h"
+#include "os_pic.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static uint8_t gButtonMask            = 0;
+static uint8_t gButtonSameMaskCounter = 0;
+
+/**********************
+ *      VARIABLES
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static nbgl_buttonEvent_t maskToEvent(uint8_t mask)
+{
+    nbgl_buttonEvent_t event = INVALID_BUTTON_EVENT;
+    switch (mask) {
+        case RELEASED_MASK | LEFT_BUTTON | RIGHT_BUTTON:
+            event = BUTTON_BOTH_PRESSED;
+            break;
+
+        case LEFT_BUTTON | RIGHT_BUTTON:
+            event = BUTTON_BOTH_TOUCHED;
+            break;
+
+        case CONTINUOUS_MASK | LEFT_BUTTON:
+            event = BUTTON_LEFT_CONTINUOUS_PRESSED;
+            break;
+
+        case RELEASED_MASK | LEFT_BUTTON:
+            event = BUTTON_LEFT_PRESSED;
+            break;
+
+        case CONTINUOUS_MASK | RIGHT_BUTTON:
+            event = BUTTON_RIGHT_CONTINUOUS_PRESSED;
+            break;
+
+        case RELEASED_MASK | RIGHT_BUTTON:
+            event = BUTTON_RIGHT_PRESSED;
+            break;
+    }
+
+    return event;
+}
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+/**
+ * @brief Function to be called periodically to check touchscreen state
+ * and coordinates
+ * @param buttonState state of both buttons (only 2 LSB are used)
+ */
+void nbgl_buttonsHandler(uint8_t buttonState)
+{
+    uint8_t button_mask;
+    uint8_t button_same_mask_counter;
+
+    // enable speeded up long push (continuous)
+    if (buttonState == gButtonMask) {
+        if (buttonState == 0) {
+            // nothing to be done when both buttons released twice in a row
+            return;
+        }
+        // each 100ms ~
+        if (gButtonSameMaskCounter < 0xff) {
```

Comment:
Is there an issue? `gButtonSameMaskCounter` type is `unint8_t`, from 0 to 0xff :thinking:  this statement is always `true`?

Reply:
`gButtonSameMaskCounter` can be equal to `0xff`? And therefore the value will stay equal to  `0xff`?